front/lib/models/data_source.ts

```diff
@@ -88,6 +91,11 @@ DataSource.init(
   }
 );
 Workspace.hasMany(DataSource, {
-  foreignKey: { allowNull: false },
+  as: "w",
+  foreignKey: { name: "workspaceId", allowNull: false },
   onDelete: "CASCADE",
 });
+DataSource.belongsTo(Workspace, {
```

Comment:
Why do we need this?

Reply:
As per https://sequelize.org/docs/v6/core-concepts/assocs/ this should not be required `belongsTo` should be used for One to One association and will create a foreign key on the source model `DataSource` which is equivalent in terms of foreign key definition to `Workspace.hasMany` which say we have a one to Many (correct here) with a foreign key on the target model.

So that belongsTo is incorrect as we are in a One-To-Many relationship.

Reply:
Hence needs to be removed :+1: